packages/amplify_core/lib/src/config/amplify_outputs/notifications/notifications_outputs.dart

```diff
@@ -0,0 +1,45 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import 'package:amplify_core/amplify_core.dart';
+import 'package:amplify_core/src/config/amplify_outputs/notifications/amazon_pinpoint_channel.dart';
+
+part 'notifications_outputs.g.dart';
+
+/// {@template amplify_core.amplify_outputs.notifications_outputs}
+/// The notifications outputs used by Notifications category plugin
+/// to communicate with backend services.
+/// {@endtemplate}
+@zAmplifyOutputsSerializable
+class NotificationsOutputs
+    with AWSEquatable<NotificationsOutputs>, AWSSerializable, AWSDebuggable {
+  /// {@macro amplify_core.amplify_outputs.notifications_outputs}
+  const NotificationsOutputs({
+    required this.awsRegion,
+    required this.amazonPinpointAppId,
+    required this.channels,
+  });
+
+  factory NotificationsOutputs.fromJson(Map<String, Object?> json) =>
+      _$NotificationsOutputsFromJson(json);
+
+  /// The AWS region of Amazon Pinpoint resources.
+  final String awsRegion;
+
+  /// Amazon Pinpoint App ID.
+  final String amazonPinpointAppId;
```

Comment:
To use the AmazonPinpoint type, we must adjust the fromJson and toJson methods to use our custom implementation instead of _$NotificationsOutputsFromJson and _$NotificationsOutputsToJson. my preference leans towards using the generated code and hence not using the AmazonPinpoint type.